hook_functions/netif_receive_skb.c

```diff
@@ -0,0 +1,84 @@
+#define TCP_PROTOCOL_CODE 6
+#define UDP_PROTOCOL_CODE 17
+
+// linux arphdr doesn't include the mac addresses, therefore we declare a new struct
```

Comment:
and `arp_hdr` function returns a pointer to a struct that contains them?

Reply:
this is how the arphdr struct defined in if_arp.h:
![image](https://user-images.githubusercontent.com/22766102/142764013-ac79f763-607a-4f00-b9c9-a546050901e5.png)